src/main/java/com/solace/samples/java/patterns/DirectReplierNonBlocking.java

```diff
@@ -0,0 +1,122 @@
+            System.out.println("The converted message payload is : " + stringPayload);
+
+            //7-B. Create the outbound message payload.
+            final StringBuilder outboundMessageStringPayload = new StringBuilder().append("Hello! Here is a response to your message on topic : ").append(inboundMessage.getDestinationName()).append(" with incoming payload :").append(stringPayload);
+
+            //7-C. Create the outbound message with headers and payload.
+            final OutboundMessage outboundMessage = outboundMessageBuilder.build(outboundMessageStringPayload.toString());
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+            System.out.println("The outbound message is : " + outboundMessage.getPayloadAsString());
+
+            //7-D. Post the reply to the incoming message
+            replier.reply(outboundMessage);
+        };
+
+        //8. Loop to identify message discards or errors and terminate if required. This should be handled in a more resilient manner
+        System.out.println(API + " " + SAMPLE_NAME + " connected, and running.");
+        //8-A. Create a scheduler variable with the required configurations to process the message in
+        final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(2);
+
+        //Request to register an asynchronous request message handler using supplied thread executor for callbacks. This method represents push-based non-blocking interface.
+        requestReplyMessageReceiver.receiveAsync(messageHandler, executorService);
+
+        while (System.in.available() == 0 && !isShutdown) {
+            if (hasDetectedDiscard) {
+                System.out.println("*** Egress discard detected *** : " + SAMPLE_NAME + " unable to keep up with full message rate");
+                hasDetectedDiscard = false;  // only show the error once per second
+            }
+        }
+        isShutdown = true;
+        requestReplyMessageReceiver.terminate(500);
+        messagingService.disconnect();
+        System.out.println("Main thread quitting.");
+    }
+
+    private static void setupPropertiesForConnection(final Properties properties, final String... args) {
+        properties.setProperty(SolaceProperties.TransportLayerProperties.HOST, args[0]);          // host:port
+        properties.setProperty(SolaceProperties.ServiceProperties.VPN_NAME, args[1]);     // message-vpn
+        properties.setProperty(SolaceProperties.AuthenticationProperties.SCHEME_BASIC_USER_NAME, args[2]);      // client-username
+        if (args.length > 3) {
+            properties.setProperty(SolaceProperties.AuthenticationProperties.SCHEME_BASIC_PASSWORD, args[3]);  // client-password
+        }
+        properties.setProperty(SolaceProperties.ServiceProperties.RECEIVER_DIRECT_SUBSCRIPTION_REAPPLY, "true");  // subscribe Direct subs after reconnect
```

Comment:
remove if default is already `true`. In 99% of cases this shouldn't need to be changed

Reply:
The java api refers to the JCSMP property Subscriptions reapply, which is by default set as FALSE.